app/src/main/java/com/example/habitsmasher/ui/history/HabitEventViewFragment.java

```diff
@@ -38,6 +61,13 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         // Inflate the layout for this fragment
         View view = inflater.inflate(R.layout.fragment_habit_event_view, container, false);
 
+        // only configure the map if the said habit event had a location saved
+        if (!_habitEvent.getLocation().equals("")) {
+            _mapView = (MapView) view.findViewById(R.id.view_map);
+            _mapView.onCreate(mapViewBundle);
+            _mapView.getMapAsync(this);
+        }
+
```

Comment:
Only display the map snippet if the habit event in question has a location specified.